File: src/wfrp-helper-web/src/data/crafting/item/Item.js
import Entity from "../../../crud/Entity";
import Price from "../../world/money/Price";

class Item extends Entity {
    name: string;
    description: string;
    weight: number;
    price: Price = new Price();
    availability: string = 'COMMON';


    updateWith(entity: *) {
        super.updateWith(entity);
        this.updateEntityProp(entity, 'price', () => new Price());
    }
}

export default Item;

File: src/main/java/pl/khuzzuk/wfrp/helper/model/crafting/inventory/Item.java
package pl.khuzzuk.wfrp.helper.model.crafting.inventory;

import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.Type;
import org.hibernate.validator.constraints.Length;
import pl.khuzzuk.wfrp.helper.common.EnumType;
import pl.khuzzuk.wfrp.helper.model.money.Price;
import pl.khuzzuk.wfrp.helper.repo.BaseEntity;

import javax.persistence.Entity;
import javax.persistence.Inheritance;
import javax.persistence.InheritanceType;
import javax.persistence.Table;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotNull;

@Getter
@Setter
@Entity
@Table(schema = "crafting")
@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
public abstract class Item extends BaseEntity {
    private @Length(min = 3) String name;
    private String description;
    private @Min(0) float weight;
    private Price price;
    @Type(type = EnumType.DEF)
    private @NotNull Availability availability = Availability.COMMON;

    @Override
    public String toString() {
        return name;
    }
}
